### DISCUSSION TITLE
[@types/node] ReadableStream.read does not match NodeJS docs

### DISCUSSION THREAD

**Discussion by cainlevy** [DefinitelyTyped/DefinitelyTyped#60526] [Issues with a @types package]
The type of NodeJS' ReadableStream says that `read` will return `string | Buffer`:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/5aa05a42660918cc3ad60a4ccc1197345ff47a40/types/node/globals.d.ts#L202

However, the official NodeJS docs (and my personal experience as of today) say that it can also return `null`. This appears to have been the case forever, from the earliest docs I could access in [v0.10.48](https://nodejs.org/docs/latest-v0.10.x/api/stream.html) all the way up to the current LTS [v16.15.0](https://nodejs.org/docs/latest-v16.x/api/stream.html#readablereadsize).

Is this fixable, or is it "too late" given the impact this would have on anyone with strict null checks?

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
Looking into source of this specific type, IIMO it was just a quick add on to stray away from 'any':
https://github.com/DefinitelyTyped/DefinitelyTyped/commit/47a28ab7798ff7097e22fd492065939e39b18b2d
There is no special caution here for `null` type IMO, feel free to make a change and PR
